salt/metalk8s/kubeadm/init/certs/apiserver.sls

```diff
@@ -0,0 +1,34 @@
+{%- from "metalk8s/map.jinja" import kube_api with context %}
+
+Create kube api private key:
+  x509.private_key_managed:
+    - name: {{ kube_api.cert.public_key }}
+    - mode: 600
+    - makedirs: True
+
+{#- Add FQDN of the server #}
+{%- do kube_api.cert.subjectAltName.get('DNS').insert(0, grains['fqdn']) %}
+
+{#- TODO: Not use all IPs and some logic to get service IP #}
+{%- do kube_api.cert.subjectAltName.get('IP').append(
+         salt['netaddress.list_cidr_ips'](kube_api.service_ip_range)[1]) %}
+{%- for ip in grains['ipv4'] %}
+  {%- if ip != '127.0.0.1' %}
+    {%- do kube_api.cert.subjectAltName.get('IP').append(ip) %}
+  {%- endif %}
+{%- endfor %}
+
+{#- Convert dictionary to string #}
+{%- do kube_api.cert.update(
+         {'subjectAltName': salt['metalk8s_certs.compute_subject_alt_name'](
+                              kube_api.cert.subjectAltName)}) %}
+Generate kube api certificate:
+  x509.certificate_managed:
+    {%- for key, value in kube_api.cert.items() %}
+    - {{ key }}: {{ value }}
+    {%- endfor %}
+    {#- If ca_server is empty then use local server as CA #}
```

Comment:
I'd rather have us error out in that case, or something. If configuration is incorrect, let's not be 'smart' and do something that... could be incorrect.

Reply:
Ok but that mean we have 2 choice:
1- We need to call CA states alone first to advertise the mine and after call the others certs state because we use mine in jinja so can't include all the state in one global state (it's ok for me to do that thought top.sls)
2- We use a slot so we need at least salt 2018.3 but that will maybe change in futur salt version
(3- we do a custom state that will just do the mine.get and call directly the x509 state)

Reply:
We can rely on Salt 2018.3 or later, no need to support anything earlier.

Nonetheless, yes, indeed, I guess we may need to create the CAs first as a separate stage/state, only then run this one.